### PR TITLE
feat: [MySQL] Enable backup configuration

### PR DESCRIPTION
**Pull request by jhvhs**
[#183876959](https://www.pivotaltracker.com/story/show/183876959)

### Checklist:

* [x] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: google-mysql.yml [hunk 1]
```diff
@@ -59,7 +59,8 @@ provision:
       examples:
         - db-n1-standard-2
         - db-n1-standard-16
-      pattern: ^[a-z][a-z0-9-]+$
+      <<: &gcp_name_constraint
```
Comment (zucchinidev) [inline google-mysql.yml:62]: praise: great reuse

**Review by zucchinidev — CHANGES_REQUESTED** [review]

File: google-mysql.yml [hunk 2]
```diff
@@ -135,6 +136,32 @@ provision:
...
+    details: The number of backups to retain. Setting this value to 0 disables the backups
+    default: 7
+    constraints:
+      minimum: 0
+  - field_name: backups_transaction_log_retention_days
+    type: number
+    details: |
+      The number of days to retain the transaction logs. Setting this value to 0 disables transaction log backups.
+      The transaction logs can be retained up to 7 days
+    default: 0
```
[the remark this answers is not in the document]
Comment (jhvhs) [reply google-mysql.yml:161]: Setting this value to `0` disables transaction log storage, which is the default

File: terraform/cloudsql/mysql/provision/main.tf [hunk 3]
```diff
@@ -15,6 +15,16 @@ resource "google_sql_database_instance" "instance" {
...
     }
+    backup_configuration {
+      enabled    = local.backups_enabled
+      start_time = var.backups_start_time
+      backup_retention_settings {
+        retained_backups = var.backups_retain_number
+        retention_unit   = "COUNT"
+      }
+      binary_log_enabled             = local.transaction_log_backups_enabled
+      transaction_log_retention_days = var.backups_transaction_log_retention_days
```
[the remark this answers is not in the document]
Comment (jhvhs) [reply terraform/cloudsql/mysql/provision/main.tf:26]: The instance was successfully created both types of backup disabled. This changes does not seem to be necessary.

File: google-mysql.yml [hunk 4]
```diff
@@ -135,6 +136,32 @@ provision:
     details: The id of the Google Compute Engine network to which the instance is connected.
     default: ""
     prohibit_update: true
+  - field_name: backups_start_time
+    type: string
+    details: Time indicating when the backup should start, in HH:MM format
+    default: "07:00"
+  - field_name: backups_location
```
[the remark this answers is not in the document]
Comment (jhvhs) [reply google-mysql.yml:143]: The default for CloudSQL is multi-regional, which is what you get if you leave this field empty. I'll update the documentation.

**Review by zucchinidev — APPROVED** [review]

**Pull request merged**